### ISSUE TITLE
Documentation is inconsistent with implementation

### ISSUE THREAD

**Issue by alexclarkofficial** [Blizzard/node-rdkafka#249]
The documentation lists an `error` event for the producer and for the consumer. It looks like the error event was removed in #144. Grepping around, it looks like the event still exists on the consumer (both standard and stream APIs) and for the producer only on the stream API.

```
lib/kafka-consumer-stream.js:158:      self.emit('error', err);
lib/kafka-consumer-stream.js:228:      self.emit('error', err);
lib/kafka-consumer-stream.js:269:      self.emit('error', err);
lib/kafka-consumer-stream.js:287:      self.emit('error', e);
lib/kafka-consumer.js:61:            self.emit('error', e);
lib/kafka-consumer.js:70:            self.emit('error', e);
lib/producer-stream.js:102:      this.emit('error', err);
lib/producer-stream.js:124:    this.emit('error', new Error('Invalid data. Can only produce buffers'));
```
I can open a PR correcting this documentation, but would like some clarity as to what is expected.

**Comment by webmakersteve** [comment]
As it stands right now error should probably only be emitted by streams, because it is expected to behave like that to follow other streams in node. I'll take a look at the consumer and see what causes it to emit error there and potentially change some things to make it more consistent.

**Comment by webmakersteve** [comment]
I believe this is fixed with 339b93545c80a76df4736e31c14898219758a3f2. Reopen if you disagree :)
